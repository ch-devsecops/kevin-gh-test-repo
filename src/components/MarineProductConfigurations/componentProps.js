import PropTypes from 'prop-types';

export default {
  fields: PropTypes.shape({
    data: PropTypes.shape({
      value: PropTypes.shape({
        model: PropTypes.shape({
          fields: PropTypes.shape({
            id: PropTypes.string,
            name: PropTypes.string,
            modelName: PropTypes.shape({
              value: PropTypes.string,
            }),
            configurations: PropTypes.arrayOf(
              PropTypes.shape({
                items: PropTypes.arrayOf(
                  PropTypes.shape({
                    configurationName: PropTypes.shape({
                      value: PropTypes.string,
                    }),
                    name: PropTypes.string,
                    tagline: PropTypes.shape({
                      value: PropTypes.string,
                    }),
                    modelCode: PropTypes.shape({
                      value: PropTypes.string,
                    }),
                    detIdentifier: PropTypes.shape({
                      value: PropTypes.string,
                    }),
                  }),
                ),
              }),
            ),
          }),
        }),
      }),
    }),
  }),
  params: PropTypes.shape({
    autoId: PropTypes.string,
    horizontalMargin: PropTypes.string,
    topMargin: PropTypes.string,
    bottomMargin: PropTypes.string,
  }),
  rendering: PropTypes.shape({
    componentName: PropTypes.string,
  }),
};
