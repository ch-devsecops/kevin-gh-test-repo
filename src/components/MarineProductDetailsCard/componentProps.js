import PropTypes from 'prop-types';

export default {
  fields: PropTypes.shape({
    data: PropTypes.shape({
      value: PropTypes.shape({
        model: PropTypes.shape({
          fields: PropTypes.shape({
            id: PropTypes.string,
            name: PropTypes.string,
            crankshaft: PropTypes.shape({
              name: PropTypes.string,
              crankshaftName: PropTypes.shape({
                value: PropTypes.string,
              }),
              series: PropTypes.shape({
                name: PropTypes.string,
              }),
            }),
            modelName: PropTypes.shape({
              value: PropTypes.string,
            }),
            tagline: PropTypes.shape({
              value: PropTypes.string,
            }),
            thumbnails: PropTypes.shape({
              value: PropTypes.shape({
                images: PropTypes.arrayOf(
                  PropTypes.shape({
                    name: PropTypes.string,
                    url: PropTypes.string,
                    alt: PropTypes.shape({
                      value: PropTypes.string,
                    }),
                    title: PropTypes.shape({
                      value: PropTypes.string,
                    }),
                  }),
                ),
              }),
            }),
            keyFeatures: PropTypes.shape({
              item: PropTypes.arrayOf(
                PropTypes.shape({
                  application: PropTypes.shape({
                    value: PropTypes.string,
                  }),
                }),
              ),
            }),
          }),
        }),
      }),
    }),
  }),
  params: PropTypes.shape({
    autoId: PropTypes.string,
    horizontalMargin: PropTypes.string,
    topMargin: PropTypes.string,
    bottomMargin: PropTypes.string,
  }),
  rendering: PropTypes.shape({
    componentName: PropTypes.string,
  }),
};
