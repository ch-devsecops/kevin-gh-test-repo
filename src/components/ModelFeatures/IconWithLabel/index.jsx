import React from 'react';
import PropTypes from 'prop-types';
import { Box, Icon, Image, Label, Optional } from '@honda-canada/design-system-react';
import styled from 'styled-components';
import { JSSFieldPropType } from '../../../utils/propTypes';
import themeStyles from './IconWithLabel.styles';

const Container = themeStyles.apply(Box, 'Container');
const IconLabel = themeStyles.apply(Label, 'IconLabel');
const IconContainer = styled('label')(themeStyles.get('IconContainer'));

const IconWithLabel = ({ featureName, isDarkMode, iconPosition, iconImage, ...props }) => (
  <Container iconPosition={iconPosition} mt="m" tabIndex={0} {...props}>
    <IconLabel color={isDarkMode ? 'white' : undefined}>{featureName}</IconLabel>
    <IconContainer>
      <Optional when={iconImage?.src}>
        <Image {...iconImage} width="20px" height="20px" />
      </Optional>
      <Optional when={!iconImage?.src}>
        <Icon name="success" iconSize="large" width="20px" height="20px" iconColor={isDarkMode ? 'white' : 'black'} />
      </Optional>
    </IconContainer>
  </Container>
);

IconWithLabel.propTypes = {
  featureName: PropTypes.string,
  isDarkMode: PropTypes.bool,
  iconPosition: PropTypes.string,
  iconImage: JSSFieldPropType,
};

export default IconWithLabel;
