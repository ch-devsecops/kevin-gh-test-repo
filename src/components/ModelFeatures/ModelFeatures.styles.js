import css from '@styled-system/css';
import { ThemeStyles } from '@honda-canada/design-system-react';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('Wrapper')(({ marginBottom, marginTop, backgroundColor }) =>
  css({
    textAlign: 'center',
    marginX: 'auto',
    marginBottom: `${marginBottom}px`,
    marginTop: `${marginTop}px`,
    backgroundColor,
  }),
);

themeStyles.addBaseStyles('List')(({ columnCount }) =>
  css({
    columnCount: [1, 1, columnCount || 4],
    columnGap: 'm',
    margin: 'auto',
    width: 'fit-content',
  }),
);

export default themeStyles;
