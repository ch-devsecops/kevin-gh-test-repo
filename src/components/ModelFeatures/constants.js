const columnMap = Object.freeze({
  one: 1,
  two: 2,
  three: 3,
  four: 4,
});

export default columnMap;
