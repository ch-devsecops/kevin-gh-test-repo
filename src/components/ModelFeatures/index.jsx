import React from 'react';
import { useSitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import PropTypes from 'prop-types';

import { createObjectFromProps } from '../../utils/object';
import { colourTokenForParam, mapGTMCategory } from '../../utils/sitecoreFields';
import { JSSFieldPropType } from '../../utils/propTypes';

import ModelFeaturesComponent from './ModelFeatures';

const ModelFeatures = ({ fields, params, rendering }) => {
  const { sitecoreContext: { route } = {} } = useSitecoreContext();

  if (!fields) return null;

  const { isDarkMode } = createObjectFromProps(route?.fields || {});
  const { data } = createObjectFromProps(fields);

  if (!data) return null;

  return (
    <ModelFeaturesComponent
      gtmTags={{
        title: data?.gtmTitle,
        category: mapGTMCategory(data?.gtmCategory),
        type: rendering?.componentName,
      }}
      isDarkMode={isDarkMode}
      checklistIconLocation={params?.checklistIconLocation?.toLowerCase()}
      bottomMargin={params?.bottomMargin}
      contentBgColour={colourTokenForParam[params?.contentBgColour?.toLowerCase()]}
      topMargin={params?.topMargin}
      shownInColumns={params?.shownInColumns?.toLowerCase()}
      bodyText={data?.bodyText}
      modelYear={data?.modelYear}
      title={data?.title}
      iconImage={data?.iconImage?.item}
      isAdditionalFeatures={data?.isAdditionalFeatures}
    />
  );
};

ModelFeatures.propTypes = {
  rendering: PropTypes.shape({ componentName: PropTypes.string }),
  params: PropTypes.shape({
    checklistIconLocation: PropTypes.string,
    bottomMargin: PropTypes.string,
    contentBgColour: PropTypes.string,
    topMargin: PropTypes.string,
    shownInColumns: PropTypes.string,
  }),
  fields: PropTypes.shape({
    bodyText: JSSFieldPropType,
    title: JSSFieldPropType,
    isAdditionalFeatures: PropTypes.bool,
    iconImage: JSSFieldPropType,
    modelYear: PropTypes.shape({
      fields: PropTypes.shape({
        standardFeatures: PropTypes.shape({
          items: PropTypes.arrayOf(
            PropTypes.shape({
              featureName: JSSFieldPropType,
              description: JSSFieldPropType,
            }),
          ),
        }),
        trims: PropTypes.arrayOf(
          PropTypes.shape({
            additionalFeatures: PropTypes.shape({
              items: PropTypes.arrayOf(
                PropTypes.shape({
                  featureName: JSSFieldPropType,
                  description: JSSFieldPropType,
                }),
              ),
            }),
          }),
        ),
      }),
    }),
  }),
};

export default ModelFeatures;
