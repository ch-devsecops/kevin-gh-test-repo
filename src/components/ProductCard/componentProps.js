import PropTypes from 'prop-types';

const compnentProps = {
  pricing: PropTypes.bool,
  fields: PropTypes.shape({
    modelName: PropTypes.shape({
      value: PropTypes.string,
    }),
    tagline: PropTypes.shape({
      value: PropTypes.string,
    }),
    primaryImage: PropTypes.shape({
      value: PropTypes.shape({
        images: PropTypes.arrayOf(
          PropTypes.shape({
            url: PropTypes.string,
            alt: PropTypes.shape({
              value: PropTypes.string,
            }),
          }),
        ),
      }),
    }),
    thumbnail: PropTypes.shape({
      value: PropTypes.shape({
        images: PropTypes.arrayOf(
          PropTypes.shape({
            url: PropTypes.string,
            alt: PropTypes.shape({
              value: PropTypes.string,
            }),
          }),
        ),
      }),
    }),
    modelPage: PropTypes.shape({
      field: PropTypes.shape({
        fields: PropTypes.shape({
          url: PropTypes.string,
        }),
      }),
    }),
    id: PropTypes.string,
    name: PropTypes.string,
  }),
  gtmTags: PropTypes.shape({
    title: PropTypes.string,
    compareTitle: PropTypes.string,
    seriesName: PropTypes.string,
    componentName: PropTypes.string,
    crankshaftName: PropTypes.string,
    interactionType: PropTypes.string,
    compareInteractionType: PropTypes.string,
  }),
};

export default compnentProps;
