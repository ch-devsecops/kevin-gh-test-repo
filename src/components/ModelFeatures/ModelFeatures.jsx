import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { Box, Button, H3, Markdown, Optional, Wrapper } from '@honda-canada/design-system-react';
import { compiler } from 'markdown-to-jsx';
import { getTitleComponent, stripMarkdownHeading } from '../../utils/markdown';
import columnMap from './constants';
import themeStyles from './ModelFeatures.styles';
import IconWithLabel from './IconWithLabel';

const FeaturesWrapper = themeStyles.apply(Wrapper, 'Wrapper');
const FeaturesList = themeStyles.apply(Box, 'List');

const Feature = (item, isDarkMode, checklistIconLocation, iconImage) => {
  const { featureName } = item;
  return (
    <IconWithLabel
      key={featureName}
      isDarkMode={isDarkMode}
      iconPosition={checklistIconLocation}
      iconImage={iconImage}
      {...item}
    />
  );
};

const StandardFeatures = ({
  gtmTags,
  isDarkMode,
  bottomMargin,
  contentBgColour,
  topMargin,
  shownInColumns,
  bodyText,
  modelYear,
  title,
  iconImage,
  checklistIconLocation,
  isAdditionalFeatures,
}) => {
  const { t } = useTranslation();
  const TitleComponent = useMemo(() => getTitleComponent(title, H3), [title]);

  return (
    <FeaturesWrapper
      data-gtm-title={gtmTags?.title}
      data-gtm-category={gtmTags?.category}
      data-gtm-component-type={gtmTags?.type}
      marginBottom={bottomMargin}
      marginTop={topMargin}
      backgroundColor={contentBgColour}
    >
      <TitleComponent mb="m" color={isDarkMode ? 'white' : undefined}>
        {compiler(stripMarkdownHeading(title))}
      </TitleComponent>
      <Markdown mb="l" color={isDarkMode ? 'white' : undefined}>
        {stripMarkdownHeading(bodyText)}
      </Markdown>
      <Button mb="xl" styling={isDarkMode ? 'special' : 'primary'}>
        {t('Pages.Models.Exploration.trimPackagesCompareTrimsLabel').toUpperCase()}
      </Button>
      <FeaturesList columnCount={columnMap[shownInColumns]}>
        <Optional when={!isAdditionalFeatures}>
          {modelYear?.standardFeatures?.item?.map(item => Feature(item, isDarkMode, checklistIconLocation, iconImage))}
        </Optional>
        <Optional when={isAdditionalFeatures}>
          {modelYear?.trims?.map(trim =>
            trim?.additionalFeatures?.item?.map(item => Feature(item, isDarkMode, checklistIconLocation, iconImage)),
          )}
        </Optional>
      </FeaturesList>
    </FeaturesWrapper>
  );
};

StandardFeatures.defaultProps = {
  isAdditionalFeatures: false,
};

StandardFeatures.propTypes = {
  isDarkMode: PropTypes.bool,
  /**
   * Is AdditionalFeatures: Checkbox[Shared] - When this gets checked,
   * the component should render the additional features instead of standard features
   * (Default state: Unchecked)
   */
  isAdditionalFeatures: PropTypes.bool,
  iconImage: PropTypes.shape({}),
  bodyText: PropTypes.string,
  title: PropTypes.string,
  // Model Year - CE selects the model year from the dropdown tree
  modelYear: PropTypes.shape({
    standardFeatures: PropTypes.shape({
      items: PropTypes.arrayOf(
        PropTypes.shape({
          featureName: PropTypes.string,
          description: PropTypes.string,
        }),
      ),
    }),
    trims: PropTypes.arrayOf(
      PropTypes.shape({
        additionalFeatures: PropTypes.shape({
          items: PropTypes.arrayOf(
            PropTypes.shape({
              featureName: PropTypes.string,
              description: PropTypes.string,
            }),
          ),
        }),
      }),
    ),
  }),
  checklistIconLocation: PropTypes.string,
  contentBgColour: PropTypes.string,
  bottomMargin: PropTypes.string,
  topMargin: PropTypes.string,
  shownInColumns: PropTypes.string,
  gtmTags: PropTypes.shape({
    title: PropTypes.string,
    category: PropTypes.string,
    type: PropTypes.string,
  }),
};

export default StandardFeatures;
