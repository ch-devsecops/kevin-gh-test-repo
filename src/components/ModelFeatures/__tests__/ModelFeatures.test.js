import React from 'react';
import { axe, cleanup, jssRender, MockRendering } from '../../../test-utils';
import ModelFeatures from '../index';

import { emptyMockData, mockData, mockSitecoreContext } from '../../../stories/ModelFeatures/componentProps';

afterEach(cleanup);

const mockRendering = new MockRendering('ModelFeatures');

describe('<ModelFeatures>', () => {
  it('should return null if there is no fields', () => {
    const { container } = jssRender(<ModelFeatures fields={null} />);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly for empty fields', () => {
    const { container } = jssRender(<ModelFeatures {...emptyMockData} />);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with all props', async () => {
    const { container } = jssRender(<ModelFeatures fields={mockData.fields} rendering={mockRendering} />, {
      context: { mockSitecoreContext },
    });

    expect(container).toMatchSnapshot();
  });

  test('should have no accessibility violations', async () => {
    const { container } = jssRender(<ModelFeatures fields={mockData.fields} rendering={mockRendering} />, {
      context: { mockSitecoreContext },
    });

    expect(container).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });
});
