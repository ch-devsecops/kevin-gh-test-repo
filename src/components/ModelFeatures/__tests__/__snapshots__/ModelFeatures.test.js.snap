// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModelFeatures> should have no accessibility violations 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 sc-1k44dfv-0 hAMmMY"
    data-gtm-category="Others"
    data-gtm-component-type="ModelFeatures"
    data-gtm-title=""
  >
    <h3
      class="sc-1k44dfv-0 ksmbdj"
    >
      <span>
        Model Features
      </span>
    </h3>
    <div
      class="sc-1k44dfv-0 eYqPQn"
      mb="l"
    >
      

      <p>
        description for model 
        <em>
          features
        </em>
         
        <strong>
          description
        </strong>
         with Markdown
      </p>
    </div>
    <button
      class="Button-pulw6t-0 jjIzEN"
      data-testid="dsr-button-component"
    >
      PAGES.MODELS.EXPLORATION.TRIMPACKAGESCOMPARETRIMSLABEL
    </button>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 euvztH"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard Features #1
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard asd Features #2
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard zxcqwe Features #3
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard asdzxcvbn Features #4
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard ewq321 asd fds 234 Features #5
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard Features #6
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<ModelFeatures> should render correctly for empty fields 1`] = `<div />`;

exports[`<ModelFeatures> should render correctly with all props 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 sc-1k44dfv-0 hAMmMY"
    data-gtm-category="Others"
    data-gtm-component-type="ModelFeatures"
    data-gtm-title=""
  >
    <h3
      class="sc-1k44dfv-0 ksmbdj"
    >
      <span>
        Model Features
      </span>
    </h3>
    <div
      class="sc-1k44dfv-0 eYqPQn"
      mb="l"
    >
      

      <p>
        description for model 
        <em>
          features
        </em>
         
        <strong>
          description
        </strong>
         with Markdown
      </p>
    </div>
    <button
      class="Button-pulw6t-0 jjIzEN"
      data-testid="dsr-button-component"
    >
      PAGES.MODELS.EXPLORATION.TRIMPACKAGESCOMPARETRIMSLABEL
    </button>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 euvztH"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard Features #1
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard asd Features #2
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard zxcqwe Features #3
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard asdzxcvbn Features #4
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard ewq321 asd fds 234 Features #5
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 grTzZX"
        tabindex="0"
      >
        <label
          class="Label-sc-115wxjh-0 sc-1k44dfv-0 clQPSl"
        >
          standard Features #6
        </label>
        <label
          class="sc-bdVaJa jIXUKR"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<ModelFeatures> should return null if there is no fields 1`] = `<div />`;
